crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -2187,17 +2207,40 @@ pub trait CheckpointServiceNotify {
     fn notify_checkpoint(&self) -> SuiResult;
 }
 
-/// This is a service used to communicate with other pieces of sui(for ex. authority)
+enum CheckpointServiceState {
+    Unstarted((CheckpointBuilder, CheckpointAggregator)),
+    Started,
+}
+
+impl CheckpointServiceState {
+    fn take(&mut self) -> (CheckpointBuilder, CheckpointAggregator) {
+        let mut state = CheckpointServiceState::Started;
+        std::mem::swap(self, &mut state);
+
+        match state {
+            CheckpointServiceState::Unstarted((builder, aggregator)) => (builder, aggregator),
+            CheckpointServiceState::Started => panic!("CheckpointServiceState is already started"),
+        }
+    }
+}
+
 pub struct CheckpointService {
     tables: Arc<CheckpointStore>,
     notify_builder: Arc<Notify>,
     notify_aggregator: Arc<Notify>,
     last_signature_index: Mutex<u64>,
+    // The highest sequence number that had already been built at the time CheckpointService
+    // was constructed
+    last_built_seq: CheckpointSequenceNumber,
+    // A notification for each time a new checkpoint is built.
+    cur_built_tx: watch::Sender<CheckpointSequenceNumber>,
```

Comment:
if my comment about `last_built_seq` above is correct, then it seems like *this* should be called `last_built_tx`? (and the above should be `highest_built_seq` or something like that?)

Reply:
renamed